nfs/src/nfs_inode.cpp

```diff
@@ -296,6 +296,117 @@ int nfs_inode::get_actimeo_max() const
+{
+    struct iovec io_vec[MAX_RPC_IOVEC_COUNT];
+    int idx = 0;
+    uint64_t start_off = 0;
+    uint64_t iov_length = 0;
+    std::vector<bytes_chunk> write_bc_vec;
+
+    /*
+     * Create the flush task to carry out the write.
+     */
+    struct rpc_task *flush_task = nullptr;
+
+    // Flush dirty membufs to backend.
+    for (bytes_chunk &bc : bc_vec) {
+        /*
+         * Get the underlying membuf for bc.
+         * Note that we write the entire membuf, even though bc may be referring
+         * to a smaller window.
+         */
+        struct membuf *mb = bc.get_membuf();
+        assert(mb != nullptr);
+
+        if (is_flush) {
+            /*
+             * get_dirty_bc_range() must have held an inuse count.
+             * We hold an extra inuse count so that we can safely wait for the
+             * flush in the following loop.
+             * This is needed as we drop inuse count if membuf is already being
+             * flushed by another thread or it may drop when the write_iov_callback()
+             * completes which can happen before we reach the waiting loop.
+             */
+            assert(mb->is_inuse());
+            mb->set_inuse();
+        }
+
+        /*
+         * Lock the membuf. If multiple writer threads want to flush the same
+         * membuf the first one will find it dirty and not flushing, that thread
+         * should initiate the Blob write. Others that come in while the 1st thread
+         * started flushing but the write has not completed, will find it "dirty
+         * and flushing" and they can avoid the write and optionally choose to wait
+         * for it to complete by waiting for the lock. Others who find it after the
+         * write is done and lock is released will find it not "dirty and not
+         * flushing". They can just skip.
+         * Note that we allocate the rpc_task for flush before the lock as it may
+         * block.
+         */
+        if (mb->is_flushing() || !mb->is_dirty()) {
+            mb->clear_inuse();
+
+            continue;
+        }
+
+        mb->set_locked();
+        if (mb->is_flushing() || !mb->is_dirty()) {
+            mb->clear_locked();
+            mb->clear_inuse();
+
+            continue;
+        }
+    
+        if (flush_task == nullptr) {
+            flush_task =
+                get_client()->get_rpc_task_helper()->alloc_rpc_task(FUSE_FLUSH);
+            flush_task->init_flush(nullptr /* fuse_req */, ino);
+        }
+
+        if(flush_task->rpc_add_iovector(io_vec, start_off, iov_length, write_bc_vec, idx, bc)) {
+            inflight_bytes += bc.length;
+            continue;
+        } else {
+            assert(idx > 0);
+            assert(write_bc_vec.size() > 0);
+            assert(iov_length > 0);
+
+            struct rpc_iovec *rpc_iov = new rpc_iovec(io_vec, idx, start_off, iov_length);
+            flush_task->issue_write_rpc(write_bc_vec, ino, rpc_iov);
```

Comment:
rpc_iovec should be part of the flush_task and rpc_add_iovector should add to that.
Also then issue_write_rpc() should not take any vector information. It should just issue the write for whatever is accumulated in the flush_task.